library/src/main/java/com/github/instacart/ahoy/Visit.java

```diff
@@ -69,7 +71,8 @@ public boolean isValid() {
     }
 
     public Visit withUpdatedExtraParams(Map<String, Object> extraParams) {
-        Map<String, Object> map = new HashMap<String, Object>(extraParams());
+        Map<String, Object> map = new ArrayMap<>();
+        map.putAll(extraParams());
```

Comment:
If we "default" on overriding, it is very hard to implement a functionality where all the `old` parameter values are removed. Maybe it is worth leaving replacing functionality?